spring/spring4/spring4-embedded/src/test/java/org/infinispan/spring/session/InfinispanEmbeddedSessionRepositoryTest.java

```diff
@@ -0,0 +1,89 @@
+package org.infinispan.spring.session;
+
+import org.infinispan.manager.DefaultCacheManager;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.spring.provider.SpringCache;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+@Test(testName = "spring.session.InfinispanEmbeddedSessionRepositoryTest", groups = "unit")
+public class InfinispanEmbeddedSessionRepositoryTest extends InfinispanSessionRepositoryTCK {
+
+   private EmbeddedCacheManager embeddedCacheManager;
+
+   @BeforeClass
+   public void beforeClass() {
+      embeddedCacheManager = new DefaultCacheManager();
+   }
+
+   @AfterMethod
+   public void afterMethod() {
+      embeddedCacheManager.getCache().clear();
+   }
+
+   @AfterClass
+   public void afterClass() {
+      embeddedCacheManager.stop();
+   }
+
+   @BeforeMethod
+   public void beforeMethod() throws Exception {
+      super.init();
+   }
+
+   @Override
+   protected SpringCache createSpringCache() {
+      return new SpringCache(embeddedCacheManager.getCache());
+   }
+
+   @Override
+   protected AbstractInfinispanSessionRepository createRepository(SpringCache springCache) throws Exception {
+      InfinispanEmbeddedSessionRepository sessionRepository = new InfinispanEmbeddedSessionRepository(springCache);
+      sessionRepository.afterPropertiesSet();
+      return sessionRepository;
+   }
+
+   @Test(expectedExceptions = NullPointerException.class)
+   @Override
+   public void testThrowingExceptionOnNullSpringCache() throws Exception {
+      super.testThrowingExceptionOnNullSpringCache();
+   }
+
+   @Override
+   public void testCreatingSession() throws Exception {
+      super.testCreatingSession();
+   }
+
+   @Override
+   public void testSavingSession() throws Exception {
+      super.testSavingSession();
+   }
+
+   @Override
+   public void testDeletingSession() throws Exception {
+      super.testDeletingSession();
+   }
+
+   @Override
+   public void testEvictingSession() throws Exception {
+      super.testEvictingSession();
+   }
+
+   @Override
+   public void testExtractingPrincipalWithWrongIndexName() throws Exception {
+      super.testExtractingPrincipalWithWrongIndexName();
+   }
+
+   @Override
+   public void testExtractingPrincipal() throws Exception {
+      super.testExtractingPrincipal();
+   }
+
+   @Override
+   public void testUpdatingTTLOnAccessingData() throws Exception {
+      super.testUpdatingTTLOnAccessingData();
+   }
+}
```

Comment:
No no, they are required....

TestNG doesn't support test inheritance by default. We use a lot of black magic to make it happen in core. But prefer to fix the problem where it originated, get rid of TestNG in Spring testsuite - see https://issues.jboss.org/browse/ISPN-7033